dist/functions.setup-appliance.sh

```diff
@@ -143,12 +143,27 @@ function get_hostname {
   else
     TIMEOUT=30
     while [ -z "$FQHOSTNAME" -o "$FQHOSTNAME" = "localhost" ];do
-      FQHOSTNAME=`hostname -f 2>/dev/null`
+      # Try to get the FQHN via hostname
+      HN=`hostname -f 2>/dev/null`
+      if [[ $HN =~ ^[A-Za-z0-9-]+(\.[A-Za-z0-9-]+)+$ ]];then
+        FQHOSTNAME=$HN
+      else
+        HN=`hostnamectl --static 2>/dev/null`
+        if [[ $HN =~ ^[A-Za-z0-9-]+(\.[A-Za-z0-9-]+)+$ ]];then
+          FQHOSTNAME=$HN
+        else
+          HN=`cat /etc/hostname 2>/dev/null`
+          [[ -n "$HN" ]] && FQHOSTNAME=$HN
+        fi
+      fi
       TIMEOUT=$(($TIMEOUT-1))
       [ "$TIMEOUT" -le 0 ] && break
       echo "Waiting for FQHOSTNAME ($TIMEOUT)"
       sleep 1
     done
+
+    # Set default to localhost
+    FQHOSTNAME="localhost"
```

Comment:
This would break all functions below which are checking if FQHOSTNAME is already set.
"localhost" is really the worst choice and should only be taken if we could not find anything else.

Reply:
See line 173/188 - Here it gets set to localhost if nothing else could be found. Its better to use an IP-Address than "localhost" - then the appliance could be reached at least by IP from the network and has a valid self-signed cert.